projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -5137,7 +5156,7 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
      * ```
      */
     get dataView(): any[] {
-        return this.verticalScrollContainer.igxForOf;
+        return this.verticalScrollContainer.igxForOf.map(rec => this.isGhostRecord(rec) ? rec.recordRef : rec) ;
```

Comment:
Is this mapping still necessary? We don't extract the original data from meta data records for other features (like group records, hierarchical records etc.) so unless there is some specific integration scenario that depends on the value to be extracted this is probably not needed.

Reply:
Additionally I suspect this getter is called quite a lot around the grid code/templates. With a sufficiently big data, this is basically creating and discarding a new array on each call.